internal/util/prometheus.go

```diff
@@ -0,0 +1,74 @@
+package util
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type ControllerFunctionalPrometheusMetrics struct {
+	// ConfigCounter number of post /config to proxy successfully
+	ConfigCounter *prometheus.CounterVec
+
+	// ParseCounter number of ingress analysis failure
+	ParseCounter *prometheus.CounterVec
+
+	// ConfigureDurationHistogram duration of last successful confiuration sync
+	ConfigureDurationHistogram prometheus.Histogram
+}
+
+type Success string
+
+const (
+	// ConfigSuccessTrue post-config to proxy successfully
+	ConfigSuccessTrue Success = "true"
+	// ConfigSuccessFalse post-config to proxy failed
+	ConfigSuccessFalse Success = "false"
+	// IngressParseTrue says that ingress parsed successful
+	IngressParseTrue Success = "true"
+	// IngressParseFalse ingress parsed failed
+	IngressParseFalse Success = "false"
```

Comment:
Are you sure that we need different consts for `true`s and `false`s used in different contexts?

I understand that they speak of success/failure of different operations, but right now this is just a fancy wrapper over `bool` and we can easily refactor if we ever need to extend either of these value groups.

I suggest that we have one `true` and one `false` label for all uses.

Reply:
agree that the declarative value is only true and false. 
The purpose of the constant is to make each operation meaningful similar as the definition of EnablementStatus

Reply:
changed 0a9651c8024cc08ed5dcbcae52d7be4d887b8415